src/script_interface/GlobalContext.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2020 The ESPResSo project
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "GlobalContext.hpp"
+#include "Exception.hpp"
+#include "ObjectHandle.hpp"
+#include "packed_variant.hpp"
+
+#include <utils/serialization/pack.hpp>
+
+namespace ScriptInterface {
+void GlobalContext::make_handle(ObjectId id, const std::string &name,
+                                const PackedMap &parameters) {
+  try {
+    ObjectRef so = m_node_local_context->make_shared(
+        name, unpack(parameters, m_local_objects));
+
+    m_local_objects.emplace(std::make_pair(id, std::move(so)));
+  } catch (Exception const &) {
+  }
```

Comment:
Because exceptions are are handled on the head node. Only exceptions from the script interface are caught, because we know that they also occur on the head node.

Reply:
If the mechanism with the context callbacks is not clear, it may be helpful to follow the call path for, say, `ObejctHandle::set_parameter`, to see what happens where. Basically the interface functions in `ObejctHandle`
call hooks in the context, and the context can then decide if it wants to react to the events. The `GlobalContext`
broadcasts the values to all nodes and then, e.g., sets the parameters on the local copies of the object, to keep
the copies on all nodes synchronous. The code you are citing here only runs on the remove copies, so the
exceptions that are catched are those of the remote local copies, which are redundant.